### PR TITLE
Feat navsys api

### PR DESCRIPTION
**Pull request by philippezwietering**
Splits all things in server and client side, with common code in between.

### CODE REVIEW

File: modules/NAVSYS/client/client.hh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * \license   See ../../LICENSE
+ */
+
+#pragma once
+
+#include <SFML/Network.hpp>
+#include <iostream>
+#include <string>
+
+class Client{
```
Comment (LuukSteeman) [inline modules/NAVSYS/client/client.hh]: Classes need documentation

File: modules/NAVSYS/server/server.hh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+#pragma once
+
+#include <SFML/Network.hpp>
+#include <vector>
+#include <memory>
+#include <string>
+#include <iostream>
+
+class Server{
```
Comment (LuukSteeman) [inline modules/NAVSYS/server/server.hh]: Class needs documentation

**Review by LuukSteeman — CHANGES_REQUESTED** [review]

**Comment by philippezwietering** [comment]
Added documentation on client and server classes

File: modules/NAVSYS/client/client.hh (hunk 1 above)
Comment (philippezwietering) [reply modules/NAVSYS/client/client.hh]: Done

File: modules/NAVSYS/server/server.hh (hunk 2 above)
Comment (philippezwietering) [reply modules/NAVSYS/server/server.hh]: Done

**Pull request merged**